routers/repo/view.go

```diff
@@ -238,3 +239,148 @@ func Home(ctx *middleware.Context) {
 	ctx.Data["BranchLink"] = branchLink
 	ctx.HTML(200, HOME)
 }
+
+func Language(ext string) string {
```

Comment:
No there is no auto-match to use in the file viewer. We have to provide the language ourselves to let the library know which language on the file. Prettyprint works the same if I'm not wrong and that's why it was barely working.

And I doubt it is easy to do it differently. Guessing the language from the source text would mean that the library must know all languages grammar and decide in which grammar is that code written. I bet is an almost impossible task or at least a huge task considering that even the code could have syntax errors.